go/internal/controller/translator/adk_api_translator.go

```diff
@@ -146,6 +152,57 @@ func (a *adkApiTranslator) TranslateAgent(
 	}
 }
 
+func (a *adkApiTranslator) validateAgent(ctx context.Context, agent *v1alpha2.Agent, state *tState) error {
+
+	agentRef := utils.GetObjectRef(agent)
+
+	if state.isVisited(agentRef) {
+		return fmt.Errorf("cycle detected in agent tool chain: %s -> %s", agentRef, agentRef)
+	}
+
+	if state.depth > MAX_DEPTH {
+		return fmt.Errorf("recursion limit reached in agent tool chain: %s -> %s", agentRef, agentRef)
+	}
+
+	if agent.Spec.Type != v1alpha2.AgentType_Declarative {
```

Comment:
should you check this before the previous two checks or does the max depth and cycle checks apply to declarative agents asw ell?